staging/src/k8s.io/legacy-cloud-providers/azure/clients/armclient/azure_armclient.go

```diff
@@ -119,6 +119,10 @@ func (c *Client) Send(ctx context.Context, request *http.Request) (*http.Respons
 		return response, rerr
 	}
 
+	if response == nil && rerr == nil {
+		return response, rerr
```

Comment:
This seems like a bug that should be fixed in sendRequest… clients  expect the invariant that either response or err will be non-nil.

Reply:
yea. let's add the workaround here and check whether could we fix the underlying SDK bug.

Reply:
@jhendrixMSFT do you know something about this? I think it's probably something wrong from go-autorest.

Reply:
I meant we should fix the bug in Client#sendRequest in this file

Reply:
There was a bug I recently fixed that was causing a nil response and error; this was a corner-case in authentication, see https://github.com/Azure/go-autorest/issues/547 for the details.  I can't tell from the stack if auth is relevant here so it might be unrelated.

Reply:
@liggitt understood, makes sense, updated.
@jhendrixMSFT I can provide the upper stack frames, would that help? Context was fuzzy: cluster-autoscaler was frequently restarting and otherwise logging many of http 429/"`client throttled`". It sent a few requests before crashing (cf. the "Invalidating unowned instance cache" log entry just before).

Reply:
Any additional data/frames/etc you have would be great.

Reply:
Edited the PR description, now with the full stack trace.